packages/pangea-sdk/src/main/java/cloud/pangeacyber/pangea/file_scan/FileScanClient.java

```diff
@@ -96,13 +112,45 @@ public FileScanClient build() {
 	 */
 	public FileScanResponse scan(FileScanRequest request, File file) throws PangeaException, PangeaAPIException {
 		FileScanFullRequest fullRequest;
-		if (request.getTransferMethod() == TransferMethod.DIRECT) {
-			FileParams fileParams = Utils.getFSparams(file);
+		TransferMethod tm = request.getTransferMethod();
+		String name;
+
+		if (tm == TransferMethod.PUT_URL) {
+			throw new PangeaException(String.format("%s not supported. User GetUploadURL() instead", tm), null);
+		}
+
+		if (tm == TransferMethod.DIRECT || tm == TransferMethod.POST_URL) {
+			FileParams fileParams = Utils.getFileUploadParams(file);
 			fullRequest = new FileScanFullRequest(request, fileParams);
+			name = "file";
 		} else {
 			fullRequest = new FileScanFullRequest(request);
+			name = "upload";
+		}
+
+		FileData fileData = new FileData(file, name);
+		return post("/v1/scan", fullRequest, fileData, FileScanResponse.class);
+	}
+
+	public AcceptedResponse requestUploadURL(FileScanUploadURLRequest request)
+		throws PangeaException, PangeaAPIException {
+		TransferMethod tm = request.getTransferMethod();
+		if (tm == null) {
+			tm = TransferMethod.PUT_URL;
+		}
+
+		if (tm == TransferMethod.MULTIPART) {
+			throw new PangeaException(String.format("%s not supported. User scan() instead", tm), null);
```

Comment:
Looks like an extra `r`. Should be `Use scan() instead`.

Reply:
Fixed